devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -0,0 +1,507 @@
+    }
+
+    preferences {
+        configParams.each {
+            if (it.name) {
+                if (it.range) {
+                    input "configParam${it.num}", "number",
+                            title: "${it.name}:",
+                            required: false,
+                            defaultValue: "${it.value}",
+                            range: it.range
+                }
+                else {
+                    input "configParam${it.num}", "enum",
+                            title: "${it.name}:",
+                            required: false,
+                            defaultValue: "${it.value}",
+                            options: it.options
+                }
+            }
+        }
+    }
+}
+
+def installed() {
+    logDebug "installed()..."
+    sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+    state.refreshConfig = true
+}
+
+private static def getCheckInterval() {
+    // These are battery-powered devices, and it's not very critical
+    // to know whether they're online or not – 12 hrs
+    return (60 * 60 * 3) + (5 * 60)
+}
+
+def updated() {
+    if (!isDuplicateCommand(state.lastUpdated, 5000)) {
+        state.lastUpdated = new Date().time
+
+        logDebug "updated()..."
+        if (device.latestValue("checkInterval") != checkInterval) {
+            sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+        }
+
+        runIn(5, executeConfigureCmds, [overwrite: true])
+    }
+
+    return []
+}
+
+def configure() {
+    logDebug "configure()..."
+
+    if (state.resyncAll == null) {
+        state.resyncAll = true
+        runIn(8, executeConfigureCmds, [overwrite: true])
+    } else {
+        if (!pendingChanges) {
+            state.resyncAll = true
+        }
+        executeConfigureCmds()
+    }
+    return []
+}
+
+def executeConfigureCmds() {
+    runIn(6, refreshSyncStatus)
+
+    def cmds = []
+
+    configParams.each { param ->
+        def storedVal = getParamStoredValue(param.num)
+        def paramVal = param.value
+        if (state.resyncAll || ("${storedVal}" != "${paramVal}")) {
+            cmds << secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: paramVal))
+            if (param.num == temperatureReportThresholdParam.num) {
+                cmds << "delay 3000"
+                cmds << secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: 0, sensorType: tempSensorType))
+            }else{
+                cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+            }
+        }
+    }
+
+    state.resyncAll = false
+    def cmds1 =delayBetween(cmds, 500)
+    def actions = []
+    cmds1.each {
+        actions << new physicalgraph.device.HubAction(it)
+    }
+    sendHubCommand(actions, 100)
```

Comment:
I think that 
```
sendHubCommands(cmds, 500)
```
would be sufficient here, but please verify it (and change it, if that's true).

Reply:
The function "sendHubCommands" is a custom function and is only called once.  I just copied the function body to the place of call.

Reply:
`sendHubCommand` isn't only a custom function - it is available for you right away. 
If you call it like this
```
def cmds = [List, of, commands]
sendHubCommand(cmds, delayBetweenCommands)
```
it'll be sufficient. You don't even have to call `physicalgraph.device.HubAction(cmd)`.

Reply:
OK, I have changed it as you said.